core/domain/topic_services.py

```diff
@@ -0,0 +1,175 @@
+    """
+
+    return topic_domain.TopicRights(
+        topic_rights_model.id,
+        topic_rights_model.manager_ids
+    )
+
+
+def save_topic_rights(topic_rights, committer_id, commit_message, commit_cmds):
+    """Saves a TopicRights domain object to the datastore.
+
+    Args:
+        topic_rights: TopicRights. The rights object for the given
+            topic.
+        committer_id: str. ID of the committer.
+        commit_message: str. Descriptive message for the commit.
+        commit_cmds: list(dict). A list of commands describing what kind of
+            commit was done.
+    """
+
+    model = topic_models.TopicRightsModel.get(topic_rights.id, strict=False)
+
+    model.manager_ids = topic_rights.manager_ids
+
+    model.commit(committer_id, commit_message, commit_cmds)
+
+
+def create_new_topic_rights(topic_id, committer_id):
+    """Creates a new topic rights object and saves it to the datastore.
+
+    Args:
+        topic_id: str. ID of the topic.
+        committer_id: str. ID of the committer.
+    """
+
+    topic_rights = topic_domain.TopicRights(topic_id, [])
+    commit_cmds = [{'cmd': topic_domain.CMD_CREATE_NEW}]
+
+    topic_models.TopicRightsModel(
+        id=topic_rights.id,
+        manager_ids=topic_rights.manager_ids
+    ).commit(committer_id, 'Created new topic rights', commit_cmds)
+
+
+def get_topic_rights(topic_id, strict=True):
+    """Retrieves the rights object for the given topic.
+
+    Args:
+        topic_id: str. ID of the topic.
+
+    Returns:
+        TopicRights. The rights object associated with the given topic.
+
+    Raises:
+        EntityNotFoundError. The topic with ID topic_id was not
+            found in the datastore.
+    """
+
+    model = topic_models.TopicRightsModel.get(topic_id, strict=strict)
+
+    if model is None:
+        return None
+
+    return get_topic_rights_from_model(model)
+
+
+def check_can_manage_topic(user, topic_rights):
+    """Checks whether the user can manage the given topic.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        topic_rights: TopicRights or None. Rights object for the given topic.
+
+    Returns:
+        bool. Whether the given user can manage the given topic.
+    """
+
+    if topic_rights is None:
+        return False
+    if role_services.ACTION_EDIT_OWNED_TOPIC not in user.actions:
+        return False
+    if topic_rights.is_manager(user.user_id):
+        return True
+
+    return False
+
+
+def assign_role(committer, assignee_id, new_role, topic_id):
+    """Assigns a new role to the user.
+
+    Args:
+        committer: UserActionsInfo. UserActionInfo object for the user
+            who is performing the action.
+        assignee_id: str. ID of the user whose role is being changed.
+        new_role: str. The name of the new role
+            ROLE_MANAGER
+            (TODO: Add more roles here if required)
```

Comment:
Drop this, it's not specific/actionable. TODOs should generally explain specifically what needs to be done. Since we don't even know the requirements yet, it's fine to remove this; it'll just get added later as a new feature if needed.

Reply:
Done